src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenNullable.vb

```diff
@@ -4831,5 +4788,2021 @@ VerifyIL("Module1.Main",
+  IL_000d:  brtrue.s   IL_003c
+  IL_000f:  call       "Function Module1.GetBool3() As Boolean?"
+  IL_0014:  stloc.1
+  IL_0015:  ldloca.s   V_1
+  IL_0017:  call       "Function Boolean?.GetValueOrDefault() As Boolean"
+  IL_001c:  brtrue.s   IL_003c
+  IL_001e:  call       "Function Module1.GetBool3() As Boolean?"
+  IL_0023:  stloc.1
+  IL_0024:  ldloca.s   V_1
+  IL_0026:  call       "Function Boolean?.GetValueOrDefault() As Boolean"
+  IL_002b:  brtrue.s   IL_003c
+  IL_002d:  call       "Function Module1.GetBool3() As Boolean?"
+  IL_0032:  stloc.1
+  IL_0033:  ldloca.s   V_1
+  IL_0035:  call       "Function Boolean?.GetValueOrDefault() As Boolean"
+  IL_003a:  brfalse.s  IL_0040
+  IL_003c:  ldc.i4.2
+  IL_003d:  stloc.0
+  IL_003e:  br.s       IL_0042
+  IL_0040:  ldc.i4.3
+  IL_0041:  stloc.0
+  IL_0042:  ldloc.0
+  IL_0043:  ret
+}
+                ]]>)
+        End Sub
+
+        <Fact()>
+        <WorkItem(38305, "https://github.com/dotnet/roslyn/issues/38305")>
+        Public Sub BooleanExpression_29()
+            Dim verifyer = CompileAndVerify(
+                <compilation>
+                    <file name="a.vb"><![CDATA[
+Option Explicit On
+
+Module Module1
+
+    Sub Main()
+    End Sub
+
+    Function Test1() As Integer
+        If (GetBool3() OrElse GetBool3()) OrElse GetBool3() OrElse GetBool3() Then
+            Return 2
+        Else
+            Return 3
+        End If
+    End Function
+
+    Function GetBool3() As Boolean?
+        Return True
+    End Function
+
+End Module
+                    ]]></file>
+                </compilation>)
+
+            verifyer.VerifyIL("Module1.Test1",
+            <![CDATA[
+{
+  // Code size       68 (0x44)
+  .maxstack  1
+  .locals init (Integer V_0, //Test1
+                Boolean? V_1)
+  IL_0000:  call       "Function Module1.GetBool3() As Boolean?"
+  IL_0005:  stloc.1
+  IL_0006:  ldloca.s   V_1
+  IL_0008:  call       "Function Boolean?.GetValueOrDefault() As Boolean"
+  IL_000d:  brtrue.s   IL_003c
+  IL_000f:  call       "Function Module1.GetBool3() As Boolean?"
+  IL_0014:  stloc.1
+  IL_0015:  ldloca.s   V_1
+  IL_0017:  call       "Function Boolean?.GetValueOrDefault() As Boolean"
+  IL_001c:  brtrue.s   IL_003c
+  IL_001e:  call       "Function Module1.GetBool3() As Boolean?"
+  IL_0023:  stloc.1
+  IL_0024:  ldloca.s   V_1
+  IL_0026:  call       "Function Boolean?.GetValueOrDefault() As Boolean"
+  IL_002b:  brtrue.s   IL_003c
+  IL_002d:  call       "Function Module1.GetBool3() As Boolean?"
+  IL_0032:  stloc.1
+  IL_0033:  ldloca.s   V_1
+  IL_0035:  call       "Function Boolean?.GetValueOrDefault() As Boolean"
+  IL_003a:  brfalse.s  IL_0040
+  IL_003c:  ldc.i4.2
+  IL_003d:  stloc.0
+  IL_003e:  br.s       IL_0042
+  IL_0040:  ldc.i4.3
+  IL_0041:  stloc.0
+  IL_0042:  ldloc.0
+  IL_0043:  ret
+}
+                ]]>)
+        End Sub
+
+        <Fact(Skip:="Only manual execution due to time.")>
```

Comment:
>Skip [](start = 14, length = 4)

We won't notice any regressions if the test is skipped. Could the number of iterations be reduced so this test could be enabled?

Reply:
> We won't notice any regressions if the test is skipped. Could the number of iterations be reduced so this test could be enabled?

I think that IL tests that I added are sufficient to catch a regression. I used this test to prove that what we emit actually executes the way we expect.

---
In reply to: [328832948](https://github.com/dotnet/roslyn/pull/38802#discussion_r328832948) [](ancestors = 328832948)